<?xml version="1.0" encoding="UTF-8" ?>

<launch>
    <arg name="publish_data" default="true"/>
    <arg name="dataset" default="kitti" doc="supported datasets to train: [asl, kitti]"/>
    <arg name="rviz_config" default="$(dirname)/../config/train_$(arg dataset).rviz"/>

    <node name="train" pkg="depth_correction" type="train_$(arg dataset)" output="screen">
        <rosparam subst_value="true">
            publish_data: $(arg publish_data)
        </rosparam>
    </node>

    <node if="$(arg publish_data)" name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_config)"/>
</launch>
